linux/x86-64/python3/acrcloud/async_recognizer.py

```diff
@@ -0,0 +1,278 @@
+            content_type = 'multipart/form-data; boundary=%s' % boundary
+            return content_type, body
+        except Exception as e:
+            print('encode_multipart_formdata error' + str(e))
+        return None, None
+
+    async def do_recogize(self, host, query_data, query_type, access_key, access_secret, timeout=5):
+        http_method = "POST"
+        http_url_file = "/v1/identify"
+        data_type = query_type
+        signature_version = "1"
+        timestamp = int(time.mktime(datetime.datetime.utcfromtimestamp(time.time()).timetuple()))
+        sample_bytes = str(len(query_data))
+
+        string_to_sign = http_method+"\n"+http_url_file+"\n"+access_key+"\n"+data_type+"\n"+signature_version+"\n"+str(timestamp)
+        hmac_res = hmac.new(access_secret.encode('ascii'), string_to_sign.encode('ascii'), digestmod=hashlib.sha1).digest()
+        sign = base64.b64encode(hmac_res).decode('ascii')
+
+        fields = {'access_key':access_key,
+                  'sample_bytes':sample_bytes,
+                  'timestamp':str(timestamp),
+                  'signature':sign,
+                  'data_type':data_type,
+                  "signature_version":signature_version}
+
+        sample_bytes = 0
+        sample_hum_bytes = 0
+        if 'sample' in query_data:
+            if query_data['sample'] == None:
+                return ACRCloudStatusCode.get_result_error(ACRCloudStatusCode.AUDIO_ERROR_CODE)
+            sample_bytes = len(query_data['sample'])
+            fields['sample_bytes'] = str(sample_bytes)
+
+        if 'sample_hum' in query_data:
+            if query_data['sample_hum'] == None:
+                return ACRCloudStatusCode.get_result_error(ACRCloudStatusCode.AUDIO_ERROR_CODE)
+            sample_hum_bytes = len(query_data['sample_hum'])
+            fields['sample_hum_bytes'] = str(sample_hum_bytes)
+
+        if sample_bytes == 0 and sample_hum_bytes == 0:
+            return ACRCloudStatusCode.get_result_error(ACRCloudStatusCode.NO_RESULT_CODE)
+
+        server_url = 'http://' + host + http_url_file
+        res = await self.post_multipart(server_url, fields, query_data, timeout)
+        return res
+
+    async def recognize(self, wav_audio_buffer):
+        res = ''
+        try:
+            query_data = {}
+            if self.recognize_type == ACRCloudRecognizeType.ACR_OPT_REC_AUDIO or self.recognize_type == ACRCloudRecognizeType.ACR_OPT_REC_BOTH:
+                query_data['sample'] = acrcloud_extr_tool.create_fingerprint(wav_audio_buffer, False)
+
+            if self.recognize_type == ACRCloudRecognizeType.ACR_OPT_REC_HUMMING or self.recognize_type == ACRCloudRecognizeType.ACR_OPT_REC_BOTH:
+                query_data['sample_hum'] = acrcloud_extr_tool.create_humming_fingerprint(wav_audio_buffer)
+
+            res = await self.do_recogize(self.host, query_data, self.query_type, self.access_key, self.access_secret,
+                                   self.timeout)
```

Comment:
Even though you've changed the line it would probably be better not to wrap this (or if you want to re-wrap it because you've changed the line, then re-wrap it to <80 columns).